src/elastio-snap.c

```diff
@@ -1899,6 +1941,8 @@ static int file_allocate(struct file *f, uint64_t offset, uint64_t length){
 	if(ret && ret != -EOPNOTSUPP) goto error;
 	else if(!ret) goto out;
 
+	file_lock(f);
```

Comment:
What if name of the file is re-used from the previous snapshot? I mean, do we need to do unlock before `real_fallocate`?

And how the code below in this procedure with the `file_write` calls can do that after this lock?)

Reply:
Thanks for the explanation. This is the answer for the 1st question. And still it's not clear for me about
> And how the code below in this procedure with the `file_write` calls can do that after this lock?)

Reply:
Thanks, you are right. I did checkout of your branch and now I see it ) Unfortunately it wasn't clear by reading the diff only.